src/main/java/uk/gov/hmcts/darts/transcriptions/mapper/TranscriptionResponseMapper.java

```diff
@@ -233,4 +234,30 @@ private Integer getRequestorId(TranscriptionEntity transcriptionEntity) {
             return null;
         }
     }
+
+
+    public GetTranscriptionDetailResponse mapTransactionEntityToTransactionDetails(TranscriptionEntity transcriptionEntity) {
+        GetTranscriptionDetailResponse details = new GetTranscriptionDetailResponse();
+
+        if (transcriptionEntity.getTranscriptionStatus() != null) {
+            details.setTranscriptionStatusId(transcriptionEntity.getTranscriptionStatus().getId());
+        }
+
+        details.setTranscriptionId(transcriptionEntity.getId());
+        details.setIsManualTranscription(transcriptionEntity.getIsManualTranscription());
+
+        if (transcriptionEntity.getCourtCase() != null) {
+            details.setCaseNumber(transcriptionEntity.getCourtCase().getCaseNumber());
+        }
+
+        if (transcriptionEntity.getHearing() != null) {
+            details.setCourthouseId(transcriptionEntity.getHearing().getCourtroom().getCourthouse().getId());
```

Comment:
I think this logic of determining what's the courthouse id and hearing id for a transcription does not belong to a mapper. That should be encapsulated in either the `TranscriptionEntity` itself, or in a service. So we have that logic in one place and it's reusable wherever we needed it. Otherwise things will get "messy".
The same approach has been done for determining the courtcase, where TranscriptionEntity exposes a `getCourtCase()` method.

Reply:
Added to entity but I do think there is probably a better place for this logic in order to keep the entity clean and purely representing the data model. Tony mortimer and myself has talked about a DAO layer in the past which may suite this logic better.

Reply:
DAO is another option sure. I think the Service layer is good too, since we have Services atm but don't really have DAOs. 
Anywhere we can encapsulate this logic in one place only works for me.